src/dvc_task/proc/process.py

```diff
@@ -50,12 +51,17 @@ def asdict(self) -> Dict[str, Any]:
 
     def dump(self, filename: str) -> None:
```

Comment:
In `load()` above, do we need to serialize loading with `lock`/`unlock`? Are there any side-effects from `load()`? Is it possible to get rid of it?